### PR TITLE
Check interaction options to disable/enable selections and tooltip

### PR DESCRIPTION
**Pull request by Caele**
We hit a bug on our side when implementing an related feature, which overlaps with fixes for storytelling.
These changes should make the chart behave correctly in the various storytelling and sheet scenarios and similar solutions can be applied to other charts.

### CODE REVIEW

**Comment by muenva** [comment]
@Caele Could you expand a little more on what you mean by "behave correctly in the various storytelling and sheet scenarios"? What is the correct behaviour? Are there any specific scenarios you had in mind?

**Comment by Caele** [comment]
@muenva I answered something similar on slack:

> a snapshot is a json dump of that objects state (layout). Then inside storytelling you reuse that state to render the object. For extensions that means calling paint(..) with that saved layout. It also means the selections should be disabled, but some interaction can still exist when you play the story (tooltips generally).
> You can also embed whole sheets in storytelling, but they are not snapshots but live objects and should behave as such in play mode. In edit they should not be interactable.

Can also add that objects/charts can be embedded outside of the client and be fed different options regarding interaction. In addition we are looking at allowing some interaction when editing, which changes how you can check the current state (say check the sheet state (URL flag) as opposed to the object state).

The changes in this PR tries to capture all those scenarios, but some can be a bit strange. The options.tooltips/selections should be straightforward, but there are scenarios like scrolling that is more tricky. Here I've used the isAnalysis check, but might revisit that.

File: src/paint.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ function setupPaint({ $, qlik }) {
 
   return function ($element, layout) {
     // Call irregularUtils to page the data for > 10000 points
-    const maxPages = qlik.navigation.getMode() === "analysis" ? 10 : 1;
+    const maxPages = this.inAnalysisState() ? 10 : 1;
```
Comment (muenva) [inline src/paint.js]: Seems like there was another commit with changes with some of the same lines: 715f9ee But instead using `this.inEditState()`. Can we overwrite those changes with these @Caele ?

Comment (muenva) [inline src/paint.js]: `this.inAnalysisState` is a function which is ran on-demand throughout the paint several times. Is there a risk of calling it once and read it from there for this file?

Comment (Caele) [reply src/paint.js]: yes you can ignore that one

Comment (Caele) [reply src/paint.js]: you can store it at the top of the function, same as the options parameters

File: src/paint.js [hunk 2]
```diff
@@ -8,7 +8,10 @@ function setupPaint({ $, qlik }) {
...
+    const enableTooltips = this.options.tooltips;
+    const enableSelections = this.options.selections;
```
[the remark this answers is not in the document]
Comment (Caele) [reply src/paint.js]: hmm, they should not be undefined at all, i'll take a look again

[the remark this answers is not in the document]
Comment (Caele) [reply src/paint.js]: I've adjusted the code, turns out the options are not always the truth. The options "should" be constants set when creating the object, the analysis/edit states are what can change.

File: src/paint.js [hunk 3]
```diff
@@ -9,8 +9,8 @@ function setupPaint({ $, qlik }) {
   return function ($element, layout) {
     // Call irregularUtils to page the data for > 10000 points
     const maxPages = this.inAnalysisState() ? 10 : 1;
-    const enableTooltips = this.options.tooltips;
-    const enableSelections = this.options.selections;
+    const enableTooltips = this.options.tooltips !== false && this.inAnalysisState();
+    const enableSelections = this.options.selections !== false && this.inAnalysisState();
```
[the remark this answers is not in the document]
Comment (ahmed-Bazzara) [reply src/paint.js]: The thing is that, there is no selections property in options!

[the remark this answers is not in the document]
Comment (muenva) [reply src/paint.js]: Only if you are in story mode

**Comment by muenva** [comment]
original comment by @Caele:
"I made one change to the heatmap PR to fix the undef issue you saw, but I will look at making one more as our testers found something a bit odd."

**Comment by muenva** [comment]
original comment by @Caele
“Now the way to check interaction states should be ok, so checking tooltip/selections on options together with inAnalysis should be good enough for storytelling.”

**Pull request merged**